modules/javafx.graphics/src/test/java/test/javafx/scene/image/PixelBufferTest.java

```diff
@@ -0,0 +1,243 @@
+            return new Rectangle2D(1, 1, WIDTH - 1, HEIGHT - 1);
+        };
+        pb.updateBuffer(callback);
+    }
+
+    @Test
+    public void testUpdatePixelBufferCallbackNull() {
+        try {
+            PixelBuffer<ByteBuffer> pb = new PixelBuffer<>(WIDTH, HEIGHT, byteBuffer, byteBGRAPrePf);
+            pb.updateBuffer(null);
+            fail("Expected NullPointerException");
+        } catch (NullPointerException e) {
+        }
+    }
+
+    @Test
+    public void testCreatePixelBufferIntArgbPrePF() {
+        PixelBuffer<IntBuffer> pb = new PixelBuffer<>(WIDTH, HEIGHT, intBuffer, intARGBPrePf);
+        assertEquals(WIDTH, pb.getWidth());
+        assertEquals(HEIGHT, pb.getHeight());
+        assertSame(intBuffer, pb.getBuffer());
+        assertSame(intARGBPrePf, pb.getPixelFormat());
+    }
+
+    @Test
+    public void testCreatePixelBufferIntArgbPreLessCapacityBuffer() {
+        try {
+            PixelBuffer<? extends IntBuffer> pb = new PixelBuffer<>(WIDTH, HEIGHT + 1, intBuffer, intARGBPrePf);
+            fail("Expected IllegalArgumentException");
+        } catch (IllegalArgumentException e) {
+        }
+    }
+
+    @Test
+    public void testCreatePixelBufferIntArgbPreDimensionsOverflow() {
+        try {
+            PixelBuffer<IntBuffer> pb = new PixelBuffer<>(0xFFFFF, 0xFFFFF, intBuffer, intARGBPrePf);
+            fail("Expected IllegalArgumentException");
+        } catch (IllegalArgumentException e) {
+        }
+    }
+
+    @Test
+    public void testCreatePixelBufferIntArgbPreWxHIsMaxint() {
+        try {
+            PixelBuffer<IntBuffer> pb = new PixelBuffer<>(0x8000, 0x10000, intBuffer, intARGBPrePf);
+            fail("Expected IllegalArgumentException");
+        } catch (IllegalArgumentException e) {
+        }
+    }
+
+    @Test
+    public void testCreatePixelBufferIntArgbPF() {
+        try {
+            PixelBuffer<IntBuffer> pb = new PixelBuffer<>(WIDTH, HEIGHT, intBuffer, PixelFormat.getIntArgbInstance());
+            fail("Expected IllegalArgumentException");
+        } catch (IllegalArgumentException e) {
+        }
+    }
+
+    @Test
+    public void testCreatePixelBufferByteBgraPF() {
+        try {
+            PixelBuffer<ByteBuffer> pb = new PixelBuffer<>(WIDTH, HEIGHT, byteBuffer, PixelFormat.getByteBgraInstance());
+            fail("Expected IllegalArgumentException");
+        } catch (IllegalArgumentException e) {
+        }
+    }
+
+    @Test
+    public void testCreatePixelBufferByteRgbPF() {
+        try {
+            PixelBuffer<ByteBuffer> pb = new PixelBuffer<>(WIDTH, HEIGHT, byteBuffer, PixelFormat.getByteRgbInstance());
+            fail("Expected IllegalArgumentException");
+        } catch (IllegalArgumentException e) {
+        }
+    }
+
+    @Test
+    public void testCreatePixelBufferIntBufferByteBgraPrePF() {
+        try {
+            PixelBuffer pb = new PixelBuffer(WIDTH, HEIGHT, intBuffer, byteBGRAPrePf);
```

Comment:
Since we are testing the raw constructor on purpose, we need to suppress the warning with `@SuppressWarnings({"rawtypes", "unchecked"})`. "rawtypes" is an Eclipse compiler warning (it's more specific than the javac compiler's "unchecked"), so I don't know what the rules are here in regards to satisfying Eclipse. javac just ignores it.

Same goes for the other test.

(`PixelBuffer pb` is also unused here)

Reply:
Removed all LHS to avoid unused warning.
I am also not sure about adding `rawtypes`. JavaFX source/tests use only `@SuppressWarnings("unchecked")`
So adding only `@SuppressWarnings("unchecked")` to keep it consistent.